examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto/gpt4all/Chat.kt

```diff
@@ -0,0 +1,51 @@
+package com.xebia.functional.xef.auto.gpt4all
+
+import arrow.core.raise.ensure
+import com.xebia.functional.gpt4all.GPT4All
+import com.xebia.functional.gpt4all.GPT4AllModel
+import com.xebia.functional.gpt4all.Message
+import com.xebia.functional.gpt4all.Response
+import com.xebia.functional.xef.AIError
+import com.xebia.functional.xef.auto.ai
+import com.xebia.functional.xef.auto.getOrThrow
+import java.nio.file.Path
+import java.util.*
+
+suspend fun main() {
+    ai {
```

Comment:
Here ai {} is just used like an effect {}. In other words, you aren't using the ai {} for prompting or something LLM related. This is a different approach than the one used with OpenAi and Hugginface

Reply:
Right, the idea is to give the option of having it as part of the AIScope in the near future, but to avoid confusion I will just wrap it on an `either` block for now.